pig/src/main/kotlin/org/partiql/pig/domain/model/TypeAnnotation.kt

```diff
@@ -0,0 +1,34 @@
+package org.partiql.pig.domain.model
+
+/**
+ * Annotations that are allowed on a UserType definition
+ *
+ * Example:
+ * ```
+ *   (sum my_sum_type
+ *       (deprecated::first_variant ...)
+ *       (second_variant ...)
+ *       (experimental::third_variant ...)
+ *   )
+ * ```
+ */
+enum class TypeAnnotation {
+    DEPRECATED,
+    EXPERIMENTAL;
+
+    companion object {
+
+        /**
+         * TODO "safe" valueOf until it's decided what to do with
+         *  -  Field identifier used in place of type name (in same sum)
+         *  -  Variant identifier used in place of type name (in a different sum)
+         *
+         *  These two TypeDomainSemanticCheckerTests appear to be asserting on missing(?) functionality
```

Comment:
What is this comment refers to?

Reply:
Those two tests in `TypeDomainSemanticCheckerTests` are asserting failure for a feature which doesn't exist. It seems like those tests were added in some previous PIG state — maybe this feature was implemented, or maybe it wasn't? The feature appears to be referencing another type within a sum definition, but that's no possible.